clients/src/main/java/org/apache/kafka/clients/producer/internals/FutureRecordMetadata.java

```diff
@@ -107,7 +108,7 @@ Long checksumOrNull() {
     RecordMetadata value() {
         if (nextRecordMetadata != null)
             return nextRecordMetadata.value();
-        return new RecordMetadata(result.topicPartition(), this.result.baseOffset(), this.relativeOffset,
+        return new RecordMetadata(result.topicPartition(), this.result.baseOffset(), this.batchIndex,
```

Comment:
The argument in `RecordMetadata` is still `relativeOffset`. Is that intended? If so, it would be good to add a comment as it's a bit confusing.

Reply:
I can also add a comment as part of https://github.com/apache/kafka/pull/10470 if that's easier (that's how I ran into this). Looking at the code, it does seem like it's intended.